### PR TITLE
69 feature implement pid for velocity control

### LINKED ISSUE #69: [Feature]: Implement PID for Velocity Control
### Description

Implement node that calculates throttle output based on current speed and speed limit.

### Definition of Done

Publishes throttle value for current speed and speed limit

### PR DESCRIPTION
**Pull request by ghost**
# Description

Implements a node that calculates and publishes a throttle output based on the current speed and speed limit using a PID-controller.

Closes #69 

## Time invested

Julian: 8 h

## Type of change

- New feature (non-breaking change which adds functionality)

## Does this PR introduce a breaking change?

This PR deletes the acting node which was used in the minimal viable vehicle (introduced in #57). The acting node is replaced by the vehicle controller (#72).

## Most important changes

Added velocity controller 

## Test

You can see this node working by merging this branch with `72-feature-implement-vehicle-controller` and enabling the VelocityPublisherDummy in the launch file.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (might be obsolete with CI later on)
- [x] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

### CODE REVIEW

**Review by erlbacsi — APPROVED** [review]
The PR handles the PID in a fully understandable way. 
Code is clear to me and commented when necessary.

time invested: 30 min

**Review by erlbacsi — CHANGES_REQUESTED** [review]
The PR handles the PID in a fully understandable way.
Code is clear to me and commented when necessary.

time invested: 30 min

File: code/acting/src/acting/velocity_controller.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                              "publish a throttle value")
+                return
+            if self.__max_velocity < 0:
+                self.logerr("Velocity controller doesn't support backward "
+                            "driving yet.")
+                raise Exception("Velocity controller doesn't support backward "
+                                "driving yet.")
+            pid.setpoint = self.__max_velocity
+            throttle = pid(self.__current_velocity)
+            throttle = max(throttle, -1.0)  # ensures that throttle >= -1
```
[the remark this answers is not in the document]
Comment (Julian-Graf) [reply code/acting/src/acting/velocity_controller.py:74]: I've made a mistake there, the CarlaEgoVehicleControlmsg only takes values in [0.0;1.0]. (https://carla.readthedocs.io/en/0.9.8/ros_msgs/#CarlaEgoVehicleControlmsg)

File: code/acting/launch/acting.launch [hunk 2]
```diff
@@ -1,10 +1,14 @@
-<!-- TODO: Insert components of acting component-->
 <launch>
     <arg name="role_name" default="ego_vehicle" />
     <arg name="control_loop_rate" default="0.1" />
 
-    <node pkg="acting" type="acting.py" name="acting" output="screen">
+    <node pkg="acting" type="velocity_controller.py" name="velocity_controller" output="screen">
```
[the remark this answers is not in the document]
Comment (Julian-Graf) [reply code/acting/launch/acting.launch:5]: When merging, make sure, that all nodes are launched in the launch file. You unfortunately have to resolve this by hand.

File: code/acting/src/acting/velocity_controller.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    """
+
+    def __init__(self):
+        super(VelocityController, self).__init__('velocity_controller')
+        self.loginfo('VelocityController node started')
+
+        self.control_loop_rate = self.get_param('control_loop_rate', 0.05)
+        self.role_name = self.get_param('role_name', 'ego_vehicle')
+
+        self.stanley_steer_sub: Subscriber = self.new_subscription(
```
[the remark this answers is not in the document]
Comment (Julian-Graf) [reply code/acting/src/acting/velocity_controller.py]: Fixed in e6ad4594c531b04114a2ea427bc4aa16e3dd9a17

File: code/acting/src/acting/velocity_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Julian-Graf) [reply code/acting/src/acting/velocity_controller.py]: Fixed in e6ad4594c531b04114a2ea427bc4aa16e3dd9a17

**Review by erlbacsi — APPROVED** [review]

**Comment by Julian-Graf** [comment]
Merge #101 first

**Pull request merged**